### DISCUSSION TITLE
Reassign the user task to supervisor

### DISCUSSION THREAD

**Discussion by Abunkar** [flowsforapex/apex-flowsforapex#713] [General]
Hi All,

I am working in flows for apex.
is this possible to reassign the user task after specific time. to his supervisor (other user). 

Example - Employee A reports to Employee B
User Task assign to A. if A did not take any action for 5 days. the task will automatically assign to User B.

Suggestions are appreciated

**Comment by rallen2010** [comment]
First question is whether you are creating your own task as an APEX page, or using an APEX Human Task (Approval Task, Action Task)?

If you are using an APEX page (i.e., a Flows for APEX user Task), the task is managed by Flows for APEX.   You could use BPMN to do reassign the task, as below...

![Screenshot 2024-05-15 at 09 48 45](https://github.com/flowsforapex/apex-flowsforapex/assets/6522588/9766bae8-530a-471f-bb41-2b79bdd89e5f)

This uses an interrupting timer boundary event on the task to cancel the original task A and then recreate it as a task for the supervisor.

We haven't added API functionality for task reassignment - it wouldn't be that difficult to do so.

If you are using APEX Human Tasks (Approval Task, Action Task), management of the task is passed to APEX.   There is functionality for a business owner to reassign tasks (delegate_task) to another user (but I don't think that you can easily set this to a timer.
